src/main/scala/scala/util/parsing/combinator/lexical/Lexical.scala

```diff
@@ -26,8 +28,11 @@ import input.CharArrayReader.EofCh
  */
 abstract class Lexical extends Scanners with Tokens {
 
+  val isLetterPattern = Pattern.compile("\\p{L}")
+
   /** A character-parser that matches a letter (and returns it).*/
-  def letter = elem("letter", _.isLetter)
+  // def letter = elem("letter", _.isLetter)
+  def letter = elem("letter", e => isLetterPattern.matcher(String.valueOf(e)).matches())
```

Comment:
These changes should live in their own commit (first commit). Btw, all other changes should live in a single (second) commit.